# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/cubeb-coreaudio/src/backend/tests/device_change.rs
# Commit: eaa23cb3b396
# Full Hash: eaa23cb3b39686e114b1e78a78c2f8e5c91bcb29
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2019-12-28 04:06:02
# Regressor Bug: 1605471
# File Overlap Count: 5
# Description:
#   Bug 1605471 - Update cubeb-coreaudio to acb90e9. r=alwu
#   
#   Pick commits:
#   acb90e9 - Add license for mixer crate
#   82e7ff9 - Move criterion crate to [dev-dependencies]
# ==============================================================================

diff -r 509cd3b3ec22 -r eaa23cb3b396 third_party/rust/cubeb-coreaudio/src/backend/tests/device_change.rs
--- a/third_party/rust/cubeb-coreaudio/src/backend/tests/device_change.rs	Fri Dec 27 16:05:26 2019 +0000
+++ b/third_party/rust/cubeb-coreaudio/src/backend/tests/device_change.rs	Fri Dec 27 16:58:18 2019 +0000
@@ -44,7 +44,7 @@
         scope
     );
 
-    let device_switcher = TestDeviceSwitcher::new(scope.clone());
+    let mut device_switcher = TestDeviceSwitcher::new(scope.clone());
 
     let count = Arc::new(Mutex::new(0));
     let also_count = Arc::clone(&count);
@@ -59,7 +59,7 @@
     test_get_started_stream_in_scope(scope.clone(), move |_stream| loop {
         thread::sleep(Duration::from_millis(500));
         changed_watcher.prepare();
-        assert!(device_switcher.next().unwrap());
+        device_switcher.next();
         changed_watcher.wait_for_change();
         if changed_watcher.current_result() >= devices.len() {
             break;
@@ -354,8 +354,8 @@
         0
     };
 
-    let input_device_switcher = TestDeviceSwitcher::new(Scope::Input);
-    let output_device_switcher = TestDeviceSwitcher::new(Scope::Output);
+    let mut input_device_switcher = TestDeviceSwitcher::new(Scope::Input);
+    let mut output_device_switcher = TestDeviceSwitcher::new(Scope::Output);
 
     test_get_stream_with_device_changed_callback(
         "stream: test callback for default device changed",
@@ -378,7 +378,7 @@
                 // switching for the default device again will be ignored.
                 while stream.switching_device.load(atomic::Ordering::SeqCst) {}
                 changed_watcher.prepare();
-                assert!(input_device_switcher.next().unwrap());
+                input_device_switcher.next();
                 changed_watcher.wait_for_change();
             }
 
@@ -387,7 +387,7 @@
                 // switching for the default device again will be ignored.
                 while stream.switching_device.load(atomic::Ordering::SeqCst) {}
                 changed_watcher.prepare();
-                assert!(output_device_switcher.next().unwrap());
+                output_device_switcher.next();
                 changed_watcher.wait_for_change();
             }
         },